browser/ui/views/playlist/playlist_action_bubble_view.cc

```diff
@@ -5,28 +5,393 @@
+    playlist::mojom::PlaylistItemPtr item,
+    base::RepeatingCallback<void(const ItemRow&)> callback)
+    : Button(base::BindRepeating(&ItemRow::OnPressed, base::Unretained(this))
+                 .Then(callback)),
+      item_(std::move(item)) {
+  auto* layout = SetLayoutManager(std::make_unique<views::BoxLayout>(
+      views::BoxLayout::Orientation::kHorizontal, gfx::Insets(8),
+      /*between_child_spacing=*/16));
+  layout->set_cross_axis_alignment(
+      views::BoxLayout::CrossAxisAlignment::kCenter);
+
+  SetPreferredSize(gfx::Size(288, 64));
+
+  auto* thumbnail = AddChildView(std::make_unique<views::ImageView>());
+  constexpr gfx::Size kThumbnailSize(64, 48);
+  // TODO(sko) We can't set the item's thumbnail for now. We need some
+  // prerequisite.
+  //  * Download the thumbnail from network
+  //  * Sanitize the image
+  thumbnail->SetImage(
+      ui::ImageModel::FromResourceId(IDR_PLAYLIST_DEFAULT_THUMBNAIL));
+  thumbnail->SetBackground(std::make_unique<DefaultThumbnailBackground>());
+  thumbnail->SetHorizontalAlignment(views::ImageViewBase::Alignment::kCenter);
+  thumbnail->SetVerticalAlignment(views::ImageViewBase::Alignment::kCenter);
+  thumbnail->SetPreferredSize(kThumbnailSize);
+  thumbnail->SetImageSize({kThumbnailSize.height(), kThumbnailSize.height()});
+
+  auto* title = AddChildView(
+      std::make_unique<views::Label>(base::UTF8ToUTF16(item_->name)));
+  title->SetHorizontalAlignment(gfx::ALIGN_LEFT);
+  layout->SetFlexForView(title, 1);
+
+  selected_icon_ = AddChildView(std::make_unique<views::ImageView>());
+  selected_icon_->SetImage(ui::ImageModel::FromVectorIcon(
+      kLeoCheckCircleOutlineIcon, kColorBravePlaylistCheckedIcon, 20));
+}
+
+int AddBubble::ItemRow::GetHeightForWidth(int width) const {
+  return GetPreferredSize().height();
+}
+
+void AddBubble::ItemRow::OnThemeChanged() {
+  views::Button::OnThemeChanged();
+  UpdateBackground();
+}
+
+AddBubble::ItemRow& AddBubble::ItemRow::OnPressed(const ui::Event& event) {
+  SetSelected(!selected_);
+  return *this;
+}
+
+void AddBubble::ItemRow::SetSelected(bool selected) {
+  if (selected_ == selected) {
+    return;
+  }
+
+  selected_ = selected;
+  selected_icon_->SetVisible(selected_);
+
+  UpdateBackground();
+}
+
+void AddBubble::ItemRow::UpdateBackground() {
+  auto* cp = GetColorProvider();
+  DCHECK(cp);
+
+  if (selected_) {
+    SetBackground(views::CreateSolidBackground(
+        cp->GetColor(kColorBravePlaylistSelectedBackground)));
+  } else {
+    SetBackground(nullptr);
+  }
+}
+
+AddBubble::AddBubble(views::View* anchor,
+                     playlist::PlaylistTabHelper* playlist_tab_helper)
+    : PlaylistActionBubbleView(anchor, playlist_tab_helper) {
+  // What this look like
+  // https://user-images.githubusercontent.com/5474642/243532255-f82fc740-eea0-4c52-b43a-378ab703d229.png
+  SetTitle(l10n_util::GetStringUTF16(IDS_PLAYLIST_ADD_TO_PLAYLIST));
+
+  SetLayoutManager(std::make_unique<views::BoxLayout>(
```

Comment:
Is there a reason you're using a BoxLayout instead of a FlexLayout here? I think it's (mostly) been replaced.

Reply:
Well, I must admit that I'm not very familiar with views::FlexLayout, so part of me hesitates to use FlexLayout 😂 The rest of me sees BoxLayout as a simplified version of FlexLayout, and views::GridLayout as an alternative to FlexLayout. To me, upstream folks don't seem to deprecate BoxLayout soon.